csrc/fusion_segmenter.cpp

```diff
@@ -3502,7 +3502,7 @@ void SegmentCandidateFinder::forwardInputs() {
       // output's further uses
       const auto& output_uses = expr->output(0)->uses();
 
-      if (output_uses.size() == 1) {
+      if (output_uses.size() == 1 && output_uses[0]->isA<UnaryOp>()) {
```

Comment:
I'm trying to understand the code around this. I don't understand why we check for `output_uses.size() == 1`.
For the test case you added, it seems fine to have T14 and T15 be the forwarded inputs (regardless of their number of uses), so we segment from them to the three outputs. What am I missing?

Reply:
Yeah, if any use is not forwarded, then we would terminate at the branching point. I don't think that necessarily has to happen though, as you say. If instead we just forward whenever all uses are UnaryOps, I think that would be fine too. We might eventually stop forwarding along each branch somewhere, but I don't really see why that would be a problem. I'll try it out soon.